File: ruby/5a.rb
#!/usr/bin/ruby

require 'digest'

input = 'cxdnnyjw'

index = 0

pass = []

while true
  # Hash each successive index
  hash = Digest::MD5.hexdigest "#{input}#{index}"
  index += 1

  # Reject invalid hashes
  next unless hash.start_with?('00000')

  # Accumulate the password
  pass << hash[5]

  # Terminate when we have the full password
  break if pass.size == 8
end

puts pass.join

File: python/5a.py
#!/usr/bin/python

from hashlib import md5
from itertools import count

INPUT = 'cxdnnyjw'

def main():
    password = []
    for index in count(0):
        hasher = md5()
        hasher.update(INPUT)
        hasher.update(str(index))
        digest = hasher.hexdigest()
        if not digest.startswith('00000'):
            continue
        password.append(digest[5])
        if len(password) == 8:
            break
    print ''.join(password)


if __name__ == '__main__':
    main()

# vi: ts=4 et:
